tuna/connection.py

```diff
@@ -174,14 +257,18 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       if abort is not None and chk_abort_file(self.id, self.logger):
         self.logger.warning('Machine %s aborted command execution: %s', self.id,
                             cmd)
-        return None, None, None
+        return i_var, o_var, e_var
 
     self.logger.error('cmd_exec retries exhausted, giving up')
-    return None, None, None
+    return i_var, o_var, e_var
 
-  def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+  def exec_command(self, cmd: str, timeout: int=int(SSH_TIMEOUT), abort: Optional[bool]=None,\
+  proc_line: Union[Any, None]= None) -> Tuple[ Union[int, Any], Union[ChannelFile, StringIO], \
+  Union[ChannelStderrFile, StringIO]]:
     # pylint: disable=too-many-nested-blocks, too-many-branches
     """Function to exec commands"""
+    o_var: ChannelFile
+    e_var: ChannelStderrFile
     _, o_var, e_var = self.exec_command_unparsed(cmd, timeout, abort)
     try:
       if not o_var:
```

Comment:
Actually this check is a result of the 'None' returns expected from exec_command_unparsed, those were replaced on 260 and 263, this 'if' and 'return' (274 & 275) may be removed.